app/src/main/java/com/github/globant/githubsubscribers/subscribersdetail/interactor/SubscriberDetailInteractorImpl.java

```diff
@@ -24,8 +25,12 @@ public void getUserData(String userName, final OnFinishedListener listener) {
         call.enqueue(new Callback<User>() {
             @Override
             public void onResponse(Call<User> call, Response<User> response) {
-                User userItem = response.body();
-                listener.onFinishedUser(userItem);
+                if (response.isSuccessful()) {
+                    User userItem = response.body();
+                    listener.onFinishedUser(userItem);
+                } else {
+                    listener.onFailureUser(Constants.MESSAGE_FAILED_SERVICE);
```

Comment:
The flow is, Interactor passes an error and type to the presenter, the presenter calls show error message on the view, the view shows the proper string.